### ISSUE TITLE
[Question] send id_token  from google js api credential to callback 

### ISSUE THREAD

**Issue by arpu** [zquestz/omniauth-google-oauth2#450]
Hi,

tested some options with the newer google api from https://developers.google.com/identity/gsi/web/reference/js-reference?hl=en

the response is jwtToken
[credential](https://developers.google.com/identity/gsi/web/reference/js-reference#credential)

How i understand the omniauth-google-oauth2 i need to POST the id_token= the JWToken ? 

and set the server option  provider_ignores_state: true  ( found in the README )

but i get 
```
self.access_token = access_token.refresh! if access_token.expired?
 ^^^^^^^^^: NoMethodError, undefined method `expired?' for nil:NilClass
```
any idea?

**Comment by zquestz** [comment]
There is a JS example at:

https://github.com/zquestz/omniauth-google-oauth2/blob/master/examples/config.ru

I believe you will want to pass `access_token`.

**Comment by arpu** [comment]
Hi @zquestz 

Thanks, this example just works with the code response from the google 

but not with the newer  credential
using POST params like 

```const req = {"access_token": response.credential}
```
error is : 

```
Nov 05 23:32:27 staging-api-cdn-west4.europe-west4-a.c.vrland-staging.internal puma[793837]: E, [2023-11-05T23:32:27.820715 #793837] ERROR -- : [6ae0e0a5-f92d-4fb7-ad16-666a7d301dcd] (google_oauth2) Authentication failure! invalid_credentials: OAuth2::Error, Invalid Value
 {
"error_description": "Invalid Value"
 }
```

setting the header  for axios 
axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest'; does not change the error

**Comment by arpu** [comment]
if i am right the new credentials should be confirmend with 

```
  payload = Google::Auth::IDTokens.verify_oidc(params[:credential], aud: ENV['GOOGLE_CLIENT_ID'])

```

described in this blog post https://patrickkarsh.medium.com/how-to-add-google-one-touch-authentication-to-a-ruby-on-rails-application-6ac8776c4190

**Comment by lhguerra** [comment]
How is this now? Is there a working example of js implementation? Also, the script in the readme is of a legacy google lib that has lost support already
![Captura de Tela 2024-08-14 às 16 21 14](https://github.com/user-attachments/assets/324dafc8-cdd0-40f2-90f3-85e48b8cb7bb)

**Comment by zquestz** [comment]
There is no working example right now. Would definitely be open to a PR to update the docs and client side JS implementation.

**Comment by lhguerra** [comment]
Please forgive me if I'm getting this wrong, but are you seriously asking the person who can't get it to work to update the docs on how to get it working?

**Comment by zquestz** [comment]
> Please forgive me if I'm getting this wrong, but are you seriously asking the person who can't get it to work to update the docs on how to get it working?

No, just saying the docs could use a new example and letting people who read this thread know it would be nice to get it updated.
